### PR TITLE
Enable full use of the BY keyword in list slices

### PR DESCRIPTION
**Pull request by Haspaker**
This commit enables full use of the `BY` keyword in list slices. Currently, `list[X to Y by Z]` only works in the current build if both `X`, `Y` and `Z` are integer literals, which is a bit confusing. E.g. `list[1 to 10 by 2]` will work while `list[1 to x by 2]` raises an error. This commit solves that.

For more information, see previous pull request #358.

Example:
```
[x, y, z] = [1 10 2]
list = [1 to 10]
list[x to y by z] #=> [ 2, 4, 6, 8, 10 ] 
```

### CODE REVIEW

**Comment by Haspaker** [comment]
The failed test is merely the one added in #641. All others work.

File: src/ast.ls [hunk 1]
```diff
@@ -2190,6 +2190,26 @@ class exports.For extends While
     head += \\n + @tab if (@item or (@index and not @object)) and \} is body.charAt 0
     head + body
 
+#### Step slice
+# Slices a list in steps
+# Makes it possible to combine non-literals and the BY keyword in slices 
+# E.g. list[1 to 10][f() to x by (1+1)]
+class exports.StepSlice extends For
+
+  makeReturn: ->
```
Comment (vendethiel) [inline src/ast.ls]: I'd rather have `(@makeReturnArg) ->`

Comment (Haspaker) [inline src/ast.ls]: Sure!

**Comment by vendethiel** [comment]
@gkz?

**Comment by gkz** [comment]
Cool, thanks!

**Pull request merged**